runtime/ops/signal.rs

```diff
@@ -60,6 +49,53 @@ impl Resource for SignalStreamResource {
   }
 }
 
+#[cfg(windows)]
+enum WindowsSignal {
+  Sigint(CtrlC),
+  Sigbreak(CtrlBreak),
+}
+
+#[cfg(windows)]
+impl From<CtrlC> for WindowsSignal {
+  fn from(ctrl_c: CtrlC) -> Self {
+    WindowsSignal::Sigint(ctrl_c)
+  }
+}
+
+#[cfg(windows)]
+impl From<CtrlBreak> for WindowsSignal {
+  fn from(ctrl_break: CtrlBreak) -> Self {
+    WindowsSignal::Sigbreak(ctrl_break)
+  }
+}
+
+#[cfg(windows)]
+impl WindowsSignal {
+  pub async fn recv(&mut self) -> Option<()> {
+    match self {
+      WindowsSignal::Sigint(ctrl_c) => ctrl_c.recv().await,
+      WindowsSignal::Sigbreak(ctrl_break) => ctrl_break.recv().await,
+    }
```

Comment:
We could also listen to `CTRL_CLOSE_EVENT` and map it to `SIGHUP`. Node does this too.